mitiq/pec/decomposition/optimal.py

```diff
@@ -0,0 +1,112 @@
+    ideal operation followed by a depolarizing channel (meaning that all
+    :math:`\mathcal{O}_{\alpha} = \mathcal{D} \circ \mathcal{U}`, where
+    :math:`\mathcal{D}(\rho) =  (1 - \epsilon) \rho + \epsilon I/2`). Given
+    that assumption, it was proven in [Takagi2020]_ that this method gives an
+    optimal decomposition for a given depolarizing ``noise_level`` (we can
+    easily calculate :math:`\epsilon` from this ``noise_level`` value). For a
+    single-qubit ``ideal_operation``, the optimal decomposition is as follows:
+
+    .. math::
+         \mathcal{U}_{\beta} = \eta_1 \mathcal{O}_1 + \eta_2 \mathcal{O}_2 +
+                               \eta_3 \mathcal{O}_3 + \eta_4 \mathcal{O}_4
+
+    .. math::
+        \eta_1 =1 + \frac{3}{4} \frac{\epsilon}{1- \epsilon},
+        \qquad \mathcal{O}_1 = \mathcal{D} \circ \mathcal{I} \circ \mathcal{U}
+
+        \eta_2 =- \frac{1}{4}\frac{\epsilon}{1- \epsilon} , \qquad
+        \mathcal{O}_2 = \mathcal{D} \circ \mathcal{X} \circ \mathcal{U}
+
+        \eta_3 =- \frac{1}{4}\frac{\epsilon}{1- \epsilon} , \qquad
+        \mathcal{O}_3 = \mathcal{D} \circ \mathcal{Y} \circ \mathcal{U}
+
+        \eta_4 =- \frac{1}{4}\frac{\epsilon}{1- \epsilon} , \qquad
+        \mathcal{O}_4 = \mathcal{D} \circ \mathcal{Z} \circ \mathcal{U}
+
+    Args:
+        ideal_operation: The input ideal operation (gate + qubit) to decompose.
+        noise_level: The noise level (as a float) of the depolarizing channel.
+
+    Returns:
+        The quasi-probability representation (QPR) of the ``ideal_operation``,
+        encoded as a list of tuples, where the first element in each tuple is
+        a float coefficient, and the second element is a list of Cirq
+        ``Operation`` objects to replace the ``ideal_operation`` with.
+
+    .. note::
+        In the description we say "noisy implementable operation" but we return
+        lists of operations. This is a subtle point, relying on our assumption
+        that we can implement *any* ``ideal_operation`` followed by a single
+        depolarizing noise channel. When running on a simulator or QPU, this
+        assumption breaks down for high ``noise_level`` values.
+
+    .. [Temme2017] : Kristan Temme, Sergey Bravyi, Jay M. Gambetta,
+        "Error mitigation for short-depth quantum circuits,"
+        *Phys. Rev. Lett.* **119**, 180509 (2017),
+        (https://arxiv.org/abs/1612.02058).
+
+    .. [Takagi2020] : Ryuji Takagi,
+        "Optimal resource cost for error mitigation,"
+        (https://arxiv.org/abs/2006.12509).
+    """
+    qubits = ideal_operation.qubits
+
+    # the single-qubit case: linear combination of 1Q Paulis
+    if len(qubits) == 1:
+        q = ideal_operation.qubits[0]
+
+        epsilon = 4 / 3 * noise_level
+        alpha_pos = 1 + ((3 / 4) * epsilon / (1 - epsilon))
+        alpha_neg = -(1 / 4) * epsilon / (1 - epsilon)
+
+        alphas = [alpha_pos] + 3 * [alpha_neg]
+        post_ops = [[P(q)] for P in PAULIS]
```

Comment:
Here I think everything works if we assume that the identity operation is noiseless. However many simulators (e.g. Cirq) add noise even to identity operations. So maybe it is more safe to "do nothing" for `alpha_pos` instead of applying the `I` gate.